File: arrays backup/Sortings/selection sort/selection sort.cpp
#include <iostream>

void insertionSort(int *array, int len) {
    for (int j = 0, min, minIndex, num; j < len; j++) {
        for (int i = j; i < len; i++) {
            if (i == j || min > array[i]) {
                min = array[i];
                minIndex = i; 
            }
        }
        num = array[minIndex];
        array[minIndex] = array[j];
        array[j] = num;
    }
}



int main()
{

    const int LEN = 8;
    int array[] = {10, 3, 6, 7, 8, 3, 1, 4};

    insertionSort(&array[0], LEN);

    for (int i = 0; i < LEN; i++)
        std::cout << array[i]<< " ";

    return 0;
}

File: arrays backup/Sortings/selection sort/selection_sort.py
def selection_sort(arr):
    j = 0
    while j < len(arr):
        min = None
        for i in range(j, len(arr)):  # current num index
            if min == None or min[0] > arr[i]:
                min = [arr[i], i]  # min Number and index of it
        arr[j], arr[min[1]] = arr[min[1]], arr[j]
        j += 1
    return arr

# Driver code
to_sort = [10, 12, 5, 8, 7, 14, 2, 1, 0]
print(selection_sort(to_sort))
